controller/sync_tasks.go

```diff
@@ -0,0 +1,127 @@
+package controller
+
+import (
+	"strings"
+
+	. "github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+)
+
+// kindOrder represents the correct order of Kubernetes resources within a manifest
+var syncPhaseOrder = map[SyncPhase]int{
+	SyncPhasePreSync:  -1,
+	SyncPhaseSync:     0,
+	SyncPhasePostSync: 1,
+}
+
+// kindOrder represents the correct order of Kubernetes resources within a manifest
+var kindOrder = map[string]int{
```

Comment:
Assigning these kinds to arbitrary numbers seems more error-prone/unmaintainable than the ordered list.  We took the sort logic (and contributed back to) Helm, so making this change makes it harder to get the upstream fixes. Two examples that illustrate my point:

1) I believe you have a bug to have "ResourceQuota"  be assigned the value -25 instead of -24.
2) we currently are missing PodDisruptionBudget from the upstream sort order, which we need to insert after PodSecurityPolicy. So now everything before needs to adjust their numbers.

Reply:
ok, I'll generate this list

Reply:
I've updated the list and the code.